channels/api_test.py

```diff
@@ -38,7 +41,15 @@
 @pytest.fixture()
 def mock_get_client(mocker):
     """Mock reddit get_client"""
-    return mocker.patch('channels.api._get_client', autospec=True)
+    mocked = mocker.patch('channels.api._get_client', autospec=True)
+    mocked.return_value.subreddit.return_value.submit.return_value.id = 'abc'
+    mocked.return_value.submission.return_value.reply.return_value.id = '456'
+    mocked.return_value.submission.return_value.reply.return_value.submission.id = '123'
+    mocked.return_value.submission.return_value.reply.return_value.subreddit.display_name = 'subreddit'
+    mocked.return_value.comment.return_value.reply.return_value.id = '789'
+    mocked.return_value.comment.return_value.reply.return_value.submission.id = '687'
+    mocked.return_value.comment.return_value.reply.return_value.subreddit.display_name = 'other_subreddit'
```

Comment:
This is a difficult testing pattern to read and maintain. Could we look into using factory boy for something like this? If that's too involved or not the right tool, I don't know if this is _much_ better, but maybe something like this?

```python
Mock = mocker.Mock
mock_client = Mock(
  subreddit=Mock(submit=Mock(id='abc')),
  submission=Mock(reply=Mock(return_value=Mock(
      id='456'
      submission=Mock(id='123', display_name='subreddit')))
  #...
)
mocked = mocker.patch('channels.api._get_client', autospec=True, return_value=mock_client)
```

You could also use `PropertyMock` to patch the `reddit` property instead of patching the `_get_client` function ([doc](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.PropertyMock)). Not a huge benefit but it's one less link in the chain of things you need to mock

Reply:
I'm not sure factory boy can deal with nested patches like this. I'll see about using the Mock you mentioned